src/realm/db.hpp

```diff
@@ -738,8 +771,14 @@ class Transaction : public Group {
     mutable _impl::History* m_history = nullptr;
 
     DB::ReadLockInfo m_read_lock;
+    util::Optional<DB::ReadLockInfo> m_oldest_version_not_persisted;
+    std::mutex mtx;
```

Comment:
add a comment indicating what this mutex is intended to protect?

Reply:
Done